src/collective/cover/layout.py

```diff
@@ -233,6 +242,112 @@ def update(self):
                 i += 1
 
 
+# Ok we exclude a little too much here. We exclude "-89", which is invalid, but
+# we also exclude "-abc" which IS a valid class, but it's also a silly CSS
+# class, so why bother to support it!?
+VALID_CSS_RE = re.compile('[a-zA-Z_][0-9a-zA-Z_-]*')
+
+def cssConstraint(value):
+    """Check for valid CSS names in a space separated string
+    """
+    invalidTokens = []
+    for token in value.split():
+        if not VALID_CSS_RE.match(token):
+            invalidTokens.append(token)
+
+    if invalidTokens:
+        raise Invalid(_(u"The following strings are not valid css identifiers: %s" %
+                        ' '.join(invalidTokens)))
+    return True
+
+
+class IRowConfigureForm(form.Schema):
+    """ Popout form to update row configuration (currently just classes)"""
+
+    form.mode(rownum='hidden')
+    rownum = schema.Int(
+            default=-1,
+        )
+
+    row_classes = schema.TextLine(
+            title=_(u"Row Classes"),
+            description=_(u"CSS class(es) for Cover row - space separated"),
+            constraint=cssConstraint,
+        )
+
+
+class RowConfigureForm(form.SchemaForm):
+    """ Configure settings for Cover Row
+
+    """
+    grok.context(ICover)
+    grok.name('configure-row')
+    grok.require('cmf.ModifyPortalContent')
+
+    schema = IRowConfigureForm
+    ignoreContext = True
+    name = 'configure_row'
+
+
+    def updateWidgets(self):
+        super(RowConfigureForm, self).updateWidgets()
+        # rownum should be passed in in query str...
+        rowid = self.request.form.get('rownum')
+        if rowid is None:
+            # ... Or by form submission (if an error occurred)
+            rowid = int(self.widgets["rownum"].value)
+
+        if rowid is None:
+            # redirect to View view if no rownum parameter
+            self.request.response.redirect(self.context.absolute_url())
+        else:
+            try:
+                rownum = int(rowid)
+                self.label = u"Row Settings for row %s" % rownum
+                self.widgets["rownum"].value = rownum
+
+                layout = json.loads(self.context.cover_layout)
+                # throws IndexError if bad rownum
+                existing_row_data = layout[rownum-1].get('data')
+                if existing_row_data:
+                    self.widgets["row_classes"].value = existing_row_data['view-classes']
+            except IndexError, ie:
+                # Bad rownum parameter, so redirect
+                self.request.response.redirect(self.context.absolute_url())
+
+
+    @button.buttonAndHandler(u"Cancel")
+    def handleCancel(self, action):
+        """User cancelled. Redirect back to the layout view.
+        """
+        layoutView = self.context.absolute_url() + "/@@layoutedit"
+        self.request.response.redirect(layoutView)
+
+
+    @button.buttonAndHandler(u'Save')
+    def handleApply(self, action):
```

Comment:
Follow PEP8 for the name of methods https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables